acnportal/acnsim/events/event.py

```diff
@@ -27,6 +30,38 @@ def __lt__(self, other):
         """
         return self.precedence < other.precedence
 
+    @property
+    def type(self):
+        """
+        Legacy accessor for event_type. This will be removed in a future
+        release.
+        """
+        warnings.warn("Accessor 'type' for type of Event is deprecated. "
+                      "Use 'event_type' instead.",
+                      DeprecationWarning)
+        return self.event_type
+
+    def to_dict(self, context_dict=None):
+        """ Implements BaseSimObj.to_dict. """
+        context_dict, = base.none_to_empty_dict(context_dict)
+        args_dict = {}
+
+        args_dict['timestamp'] = self.timestamp
+        args_dict['event_type'] = self.event_type
+        args_dict['precedence'] = self.precedence
+
+        return args_dict
+
+    @classmethod
+    def from_dict(cls, in_dict, context_dict=None, loaded_dict=None, cls_kwargs=None):
```

Comment:
The argument `cls_kwargs` is confusing and raises flags in type checkers. However, its use saves a lot of repeated code. We should decide if we want to keep this.

Reply:
I'm a fan of removing `cls_kwargs` and using helper functions to avoid the duplicated code. This reduces the number of arguments users who extend sim objects have to think about, and I think using helper functions is much more transparent than building cls_kwargs. @zach401 what do you think?

Reply:
We've agreed to remove cls_kwargs